src/Home.js

```diff
@@ -39,15 +41,35 @@ class Home extends Component {
     activity: PropTypes.bool.isRequired,
     history: PropTypes.shape({ length: PropTypes.number.isRequired })
       .isRequired,
+    searchQuery: PropTypes.string.isRequired,
   };
 
-  componentDidMount() {
+  state = {
+    // eslint-disable-next-line react/destructuring-assignment
```

Comment:
There's a way to do this that I think is quite cool and doesn't require disabling this check. It is also more flexible.

https://github.com/graasp/graasp-app-timer/blob/master/src/components/modes/teacher/Settings.js#L52-L57

Reply:
Credit to @solaolateju for the discovery.